src/routes/profile.router.js

```diff
@@ -0,0 +1,172 @@
+const { User } = require('../models/User.model');
+const { ApiError } = require('../exceptions/api.error');
+const { v4: uuidv4 } = require('uuid');
+const emailService = require('./email.service');
+const userService = require('./user.service');
+const bcrypt = require('bcrypt');
+const jwtService = require('./jwt.service');
+const tokenService = require('./token.service');
+const { checkRequired } = require('../utils/checkRequired');
+const { validateEmail } = require('../utils/validateEmail');
+const { validatePassword } = require('../utils/validatePassword');
+
+const register = async ({ name, email, password }) => {
+  const errors = {
+    name: checkRequired(name, 'name'),
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.name || errors.password) {
+    throw ApiError.badRequest('bad request', errors);
+  }
+
+  if (await userService.getByEmail(email)) {
+    throw ApiError.badRequest('bad request', {
```

Comment:
The method 'ApiError.badRequest' should be 'ApiError.BadRequest' to match the static method naming convention in 'ApiError'.